### ISSUE TITLE
Batch tool for linking FA history climbers to accounts

### ISSUE THREAD

**Issue by scd** [theCrag/website#558]
As part of the url query program we need to link real account to FA climbers field.  This means we will be able to query routes put up by XYZ.

There are some business rules to work out:
- commit to only having one FA
- can multiple climbers be associated with the FA, in which case which one is attributed with putting up the route

The challenge will be finding some nice UI tricks that fit on the bulk edit.

We probably want some way of claiming all your first ascents (in bulk, individually).

I need to add fields or a table to the database, but will wait until we have worked out how the UI is going to work.

**Comment by brendanheywood** [comment]
Ok so a few things have changed a little as a result of the recent mega release. Importantly we've gone with select2 instead of at.js for the editing, which to a certain extent forces our hand a little bit. But in either case it will be difficult to retain roundtrip editing of the @ link, as well as retaining the override text label eg ```"Some words"[@account]```. So perhaps we just have to let that slide? After all each time a real guide is rewritten the exact text isn't going to be recycled and the format might be changed to be more consistent, ie "B. Heywood" for all FA's and no nick names or whatever. The important thing is that the link goes to the right person.

Would it be worth storing this as the component parts instead of as a comma separated string? Certainly this would make it easier to query on what text names are already in use, their frequency, and doing batch replacement of them rather than lots of strong regex and replacing inside sql. Or at least as derived data if not the primary storage or if. I guess whatever logic we use here could also apply to 'with'.

**Comment by scd** [comment]
> Would it be worth storing this as the component parts instead of as a comma separated string

Let's wait and see if there is some feature that really needs it. I don't want to create extra work because there is a marginal improvement in a batch update. Splitting it is probably not a trivial piece of work and my gut is saying the benefits are marginal.

I think there will be a tangible benefit for searching free text names (linked names are split anyway through the mentions table). If global searches were important for this then we would have a significant performance improvement benefit. At the crag level it should be marginal.

Yes the 'with' field and the fa climbers field is pretty much identical, so what we do with one we do with the other.

**Comment by scd** [comment]
> But in either case it will be difficult to retain roundtrip editing

I agree with just letting this slide. The round trip differences are marginal anyway.

**Comment by lordyavin** [comment]
BTW. I have a working Java application based on the API that does exactly what this issue is about.
